src/com/zenuml/dsl/PsiToDslConverter.java

```diff
@@ -108,13 +109,21 @@ public void visitNewExpression(PsiNewExpression expression) {
         LOG.debug("Enter: visitNewExpression: " + expression);
         zenDsl
             .append("new ")
-            .append(expression.getClassReference().getReferenceName())
+            .append(getClassName(expression))
             .openParenthesis()
             .append(getArgs(expression.getArgumentList()))
             .closeParenthesis();
         super.visitNewExpression(expression);
     }
 
+    @NotNull
+    private String getClassName(PsiNewExpression expression) {
+        if (expression.getClassReference() != null) {
+            return expression.getClassReference().getReferenceName();
+        }
+        return withoutTypeParameter(expression.getType().getCanonicalText());
```

Comment:
```suggestion
        return expression.getAnonymousClass().getBaseClassReference().getReferenceName();
```
explicitly use AnonymousClass related oprations instead of regex replacement.

Reply:
It doesn't work with `new int[] { 1 }`